stackit/internal/services/argus/instance/resource.go

```diff
@@ -639,6 +748,42 @@ func mapACLField(aclList *argus.ListACLResponse, model *Model) error {
 	return nil
 }
 
+func mapMetricsRetentionField(r *argus.GetMetricsStorageRetentionResponse, model *Model) error {
+	if r == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+
+	if r.MetricsRetentionTimeRaw == nil || r.MetricsRetentionTime5m == nil || r.MetricsRetentionTime1h == nil {
+		return fmt.Errorf("metrics retention time is nil")
+	}
```

Comment:
Do we need to error if these are not defined in the API? Maybe we could just not map in this case

Reply:
I don't see a valid case where this would be nil. We could definitely not error, but not mapping could lead to inconsistent applies, where Terraform complains that values are still unknown after apply.

We could do something like mapping to 0, but then I would prefer just erroring

Reply:
Because the metrics storage retention is always set, starting with default values. So the get endpoint returning nil, has to be an error on the API side